include/lsst/jointcal/PhotometryTransfo.h

```diff
@@ -0,0 +1,105 @@
+// -*- LSST-C++ -*-
+#ifndef LSST_JOINTCAL_PHOTOMETRY_TRANSFO_H
+#define LSST_JOINTCAL_PHOTOMETRY_TRANSFO_H
+
+#include <iostream>
+#include <sstream>
+#include <memory>
+
+#include "lsst/jointcal/Point.h"
+
+namespace lsst {
+namespace jointcal {
+
+class Point;
+
+class PhotometryTransfoSpatiallyInvariant;
+
+/*
+ * A photometric transform, defined as a scale factor of the input calibration.
+ *
+ *     initialCalibFlux (Maggies) * transfo(x,y) -> correctedFlux (Maggies)
+ *
+ * @todo Eventually will be defined the same as PhotoCalib:
+ *     instFlux (ADU) -> flux (maggies)
+ *
+ * @seealso lsst::afw::image::PhotoCalib
+ */
+class PhotometryTransfo {
+public:
+    /// Apply the transform to instFlux at (x,y), put result in flux
+    virtual void apply(double x, double y, double instFlux, double &flux) const = 0;
```

Comment:
Good question. I was following gTransfo, but that deals with 2d Points, so I'll just return here.